strictdoc/backend/sdoc/document_reference.py

```diff
@@ -1,9 +1,15 @@
+from __future__ import annotations
+
 from typing import Optional
+from typing import TYPE_CHECKING
```

Comment:
Must this change be implemented within your overall change of introducing constants?

Reply:
Unfortunately yes.
The constants are defined as part of the specific document node class files, e.g. ReferenceType is defined in reference.py. By referencing the constants, often circular import dependencies are introduced. In combination with type hints, this gets even worse. For example Document=>Requirement=>DocumentReference=>Document was one of them. Moving the "from import" statements into local imports is not a good solution which is also faulted by lint/black.
Found the  TYPE_CHECKING fix on [Adam Johnson's Blog](https://adamj.eu/tech/2021/05/13/python-type-hints-how-to-fix-circular-imports/).

Reply:
Ok, it looks like by adding a few more imports you are hitting the dependency cycles. I have tried your branch on my machine and it looks like moving the newly introduced `RequirementFieldName` and the already existing `RESERVED_NON_META_FIELDS` and `RequirementFieldType` to `models/type_system.py` solves the cycles problem for now.

Within your branch, just move these types to `models/type_system.py` and you will no longer have to fight with the cycle.

My context: I am aware of the dependencies problem but I would like to solve it within some other work, not such a trivial introduction of enum values. For one thing, I was going to remove the `DocumentReference` very soon.

I see that you have removed the merge commit by rebasing. Could you make the next step and squash all commits into just one? (interactive rebase:`rebase -i` procedure)